### ISSUE TITLE
Test deployment with a Github Action

### ISSUE THREAD

**Issue by wehrad** [cryo-data/cryo-data#1]
I suggest adding a Github action in which the deployment of `cryo-data` would be tested through a couple of steps as described in the [current documentation](https://github.com/cryo-data/cryo-data#use):
- `conda install datalad`
- `git clone https://github.com/cryo-data/cryo-data`
- `cd cryo-data`
- `./cryo-data.sh .` 

This can be seen as very secondary but would improve the stability of `cryo-data`, meeting the usual CI/CD standards. It could e.g. spot efficiently broken URLs and associated symlinks and prevent them to be passed silently.

If you think this is a good idea, I can work on a PR.

**Comment by mankoff** [comment]
Yes we should support CI/CD and similar best-practices, but let's wait until a beta version and discuss before you write code and submit PRs. The beta version should not have any code, just a `datalad clone`. I'm not sure what you can test with this though - the URLS and link issues would only appear when you *download* the files, which is not within the scope of GitHub actions.

**Comment by wehrad** [comment]
Agree!

> link issues would only appear when you download the files, which is not within the scope of GitHub actions.

This is possible with Github actions. A small virtual environment can be setup in a Github action, then **a lot** of operations can be performed, including downloading and testing files (and other push/pull operations to/ftom target). Only thing is the files used in an action should be small so the tests don't take forever!

An example [here](https://stackoverflow.com/questions/68272303/how-can-i-use-github-actions-to-trigger-downloading-a-static-copy-of-a-website-a) of a simple file download in a Github action, the file then being used for other operations (here a simple zip).

**Comment by mankoff** [comment]
> Only thing is the files used in an action should be small so the tests don't take forever!

So we would only be able to test a very very small subset of this project.

**Comment by wehrad** [comment]
> So we would only be able to test a very very small subset of this project.

I think it would depend on how the targeted data sets are structured. If a combination of small files, then we could download only one file just to test the link.
